spec/dependency_container_spec.rb

```diff
@@ -0,0 +1,82 @@
+require "spec_helper"
+
+describe Nurse::DependencyContainer do
+  let(:container) {
+    container = Nurse::DependencyContainer.new
+
+    container.define(Hash) do
+      Hash.new
+    end
+
+    container.define(:definition) do
+      :defined_object
+    end
+  }
+
+  describe "#define" do
+    it "defines dependencies" do
+      container.define(:foo) do
+        :bar
+      end
+
+      container.get(:foo).must_equal :bar
+    end
+  end
+
+  describe "#defined?" do
+    it "returns false when dependency was not defined" do
+      container.defined?(:foo).must_equal false
+    end
+
+    it "returns true when dependency was defined" do
+      container.defined?(Hash).must_equal true
+    end
+
+    it "handles classes as dependencies keys" do
+      container.defined?(Hash).must_equal true
+    end
+
+    it "handles symbols as dependencies keys" do
+      container.defined?(:definition).must_equal true
+    end
+
+    it "handles strings as dependencies keys" do
+      container.defined?('definition').must_equal true
+    end
+
+  end
+
+  describe "#get" do
+    it "return nil when no dependency was defined" do
+      container.get(:undefined).must_equal nil
+    end
+
+    it "returns a unique instance of the dependency when it was defined" do
+      dependency = container.get(Hash)
+      dependency.must_be_instance_of(Hash)
+      container.get(Hash).must_be_same_as(dependency)
+    end
+  end
+
+  describe "#fetch" do
+    it "returns a unique instance of the dependency when it was defined" do
+      dependency = container.fetch(Hash)
+      dependency.must_be_instance_of(Hash)
+      container.fetch(Hash).must_be_same_as(dependency)
+    end
+
+    it "throws an exception when dependency is not defined" do
+      begin
+        container.fetch('undefined')
+        fail
+      rescue Nurse::DependencyContainer::UndefinedDependency => e
+        e.message.must_equal "'undefined' was not defined"
+      end
+    end
+
+    it "allows block as a fall back for the unexisting dependency" do
+      container.fetch('undefined') {|key| "#{key} undefined"}.must_equal "undefined undefined"
```

Comment:
Line is too long. [94/80]<br>Space between { and | missing.<br>Space missing inside }.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.